StoreApp/Controllers/StoreController.cs

```diff
@@ -199,6 +200,37 @@ public ActionResult DeleteProduct(ProductViewModel product)
             return RedirectToAction("Index");
         }
 
+        [HttpGet]
+        public ActionResult Buy(int id)
+        {
+            ProductViewModel productVM = webMapper.config.Map<ProductDTO, ProductViewModel>(productService.Get(id));
+
+            if (productVM == null)
+            {
+                TempData["Message"] = "Product don't add to cart";
+            }
+
+            var cart = TempData["Сart"] as List<ProductViewModel> ?? new List<ProductViewModel>();
+
+            ProductViewModel productCart = cart.FirstOrDefault(p => p.Id == id);
+
+            if (productCart != null)
+            {
+                productCart.Quantity++;
+            }
+            else
+            {
+                productVM.Quantity = 1;
+                cart.Add(productVM);
+            }
+
+            TempData["Сart"] = cart;
+
+            TempData["Message"] = "Product added to cart";
```

Comment:
Grammar mistake **"The product was added to the cart"**

Reply:
fixed